baram/ec2_manager.py

```diff
@@ -77,16 +77,21 @@ def get_default_vpc(self) -> list:
         """
         return [vpc for vpc in self.list_vpcs() if vpc['IsDefault']][0]
 
-    def get_sg_ids_with_vpc_id(self, vpc_id: str) -> list:
+    def get_sg_ids_with_vpc_id(self, vpc_id: str, default_only: bool = False) -> list:
         """
         Get security group ids of specific vpc.
 
         :param vpc_id: VpcId
+        :param default_only: return default security group id only if it's True
         :return: GroupId
         """
         sgs = self.cli.describe_security_groups()['SecurityGroups']
         try:
-            return [sg['GroupId'] for sg in sgs if sg['VpcId'] == vpc_id]
+            specific_sgs = [sg for sg in sgs if sg['VpcId'] == vpc_id]
```

Comment:
According to your logic, result value should be single variable not list. You should use proper data structure as you need.